File: fe/lib/models/account_model.dart
import 'package:cloud_firestore/cloud_firestore.dart';

class AccountModel {
  final String firebaseUid;
  final String fullName;
  final String email;
  final String provider;
  final String avatarUrl;
  final String role;

  AccountModel({
    required this.firebaseUid,
    required this.fullName,
    required this.email,
    required this.provider,
    this.avatarUrl = '',
    this.role = 'user',
  });

  Map<String, dynamic> toMap() => {
    'firebaseUid': firebaseUid,
    'fullName': fullName,
    'email': email,
    'provider': provider,
    'avatarUrl': avatarUrl,
    'role': role,
    'createdAt': FieldValue.serverTimestamp(),
  };

  factory AccountModel.fromMap(Map<String, dynamic> map) => AccountModel(
    firebaseUid: map['uid'] ?? '',
    fullName: map['fullName'] ?? '',
    email: map['email'] ?? '',
    provider: map['provider'] ?? '',
    avatarUrl: map['avatarUrl'] ?? '',
    role: map['role'] ?? 'user',
  );
}

File: be/models/account.model.js
const db = require("../configs/db");

const accSchema = new db.mongoose.Schema(
  {
    firebaseUid: { type: String, required: true, unique: true },
    fullName: { type: String },
    email: {
      type: String,
      required: true,
      unique: true,
      lowercase: true,
      trim: true,
    },
    provider: {
      type: String,
      enum: ["google.com", "facebook.com", "password"],
      required: true,
    },
    role: {
      type: String,
      enum: ["superAdmin", "admin", "user"],
      default: "user",
    },
    avatarUrl: { type: String, default: null },
    active: { type: Boolean, default: true },
    deletedAt: { type: Date, default: null },
  },
  { collection: "accounts", timestamps: true },
);

let accModel = db.mongoose.model("accModel", accSchema);
module.exports = { accModel };
